skipchain/skipchain.go

```diff
@@ -1168,41 +1268,56 @@ func (s *Service) bftForwardLinkLevel0Ack(msg []byte, data []byte) bool {
 // forwardLink receives a signature request of a newly accepted block.
 // It only needs the 2nd-newest block and the forward-link.
 func (s *Service) forwardLink(req *network.Envelope) error {
+	fsOrig, ok := req.Msg.(*ForwardSignature)
+	if !ok {
+		return errors.New("didn't get ForwardSignature message")
+	}
+
+	_, err := s.ForwardLinkHandler(fsOrig)
+	return err
+}
+
+// ForwardLinkHandler receives a forward-link signature request for a block already
+// appended to the chain (height >= 1).
+func (s *Service) ForwardLinkHandler(req *ForwardSignature) (*ForwardSignatureReply, error) {
```

Comment:
I'm lazy: can you tell me whether this method makes sure that the proposed link is actually valid, as in: the `from` and `to` are a valid pair within the restrictions given by the `BaseHeight` and `MaximumHeight`, respective to the indexes of the blocks?

Reply:
Looking at it, the BFT callback checks only against the size of the back links but the forward-link won't be updated in the db if the target height is wrong. But the BFT callback could be more clever.

Reply:
I opened an issue here:

https://github.com/dedis/cothority/issues/1912

does it make sense?

Reply:
Yes. Closing here then.